src/entities/session/model/storage.ts

```diff
@@ -53,10 +51,6 @@ function SessionStorage() {
     getSession,
```

Comment:
or move part of logic to setSession if you need to keep this in sync

Reply:
Add validation and throw Error based on isEqual(session, newSession) result

Reply:
The whole idea behind this function is to return the cached value instead of a new object reference. So if the `storage.getString` returns the same result as the `session` variable then we would like to return the 'session' variable.
That behavior is expected by React's `useSyncExternalStore` that is used in `useSession.ts`. If getState returns a new object reference that will lead to infinite rerenders.
But all in all, I agree with you that we should separate concerns. Perhaps, we can create a separate function/class resonsible for caching the stored value?